services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -196,6 +201,53 @@ async def update_project_last_change_timestamp(
     await db.update_project_last_change_timestamp(ProjectIDStr(f"{project_uuid}"))
 
 
+async def patch_project(
+    app: web.Application,
+    *,
+    user_id: UserID,
+    project_uuid: ProjectID,
+    project_patch: ProjectPatch,
+):
+    _project_patch_exclude_unset: dict[str, Any] = jsonable_encoder(
+        project_patch, exclude_unset=True, by_alias=False
+    )
+    db: ProjectDBAPI = app[APP_PROJECT_DBAPI]
+
+    # 1. Get project
+    project_db = await db.get_project_db(project_uuid=project_uuid)
+
+    # 2. Check user permissions
+    _user_project_access_rights = await db.get_user_project_access_rights(
+        user_id, project_uuid
+    )
+    if not _user_project_access_rights or not _user_project_access_rights.write:
+        raise ProjectInvalidRightsError(user_id=user_id, project_uuid=project_uuid)
+
+    # 3. If patching access rights
+    if new_prj_access_rights := _project_patch_exclude_unset.get("access_rights"):
```

Comment:
Suggestion: I guess you could use "project_patch.access_rights" here directly, no?

Reply:
Unfortunately, I can not, it the first step of this function:
```
_project_patch_exclude_unset: dict[str, Any] = jsonable_encoder(
        project_patch, exclude_unset=True, by_alias=False
    )
```
I use `exclude_unset=True` --> This returns a dictionary from the `project_patch` model.  Fields which were not explicitly set when creating the model are excluded from the returned dictionary. This is a good way how to for example differentiate whether user set the value None by him self, or it was set by default value in the model.